### PR TITLE
Added laravel/sail development environment integration

### LINKED ISSUE #58: Add a Docker image for easier development setup
Having a Docker image that can be spun up using `docker-compose` would make it easier to onboard new contributors. I didn't make one since I have no need for it myself, but help is welcome in adding a Docker setup.

The image should use PHP 7.2 for compatibility with the production setup.

### PR DESCRIPTION
**Pull request by steffendietz**
### What was done

I integrated a development environment, after seeing the previous effort was abandoned/stale.
I opted for `laravel/sail` because it is the officially supported docker based development setup.
This would close #58 and would supersede #104.

For a better overview I would now cite the addition to the `README.md`:

### Development environment

This project uses [Laravel Sail](https://laravel.com/docs/sail) to supply you with a docker based development environment.
Please take a moment to familiarize yourself with its concepts.

We also supply a small convenience shell script named `sail` in the project root which forwards all commands to the `vendor/bin/sail` command.

It also asks whether it should
- [install the composer dependencies using a container](https://laravel.com/docs/sail#installing-composer-dependencies-for-existing-projects), if not finding the sail command.
- install `.env.sail` as the current `.env`, if none is currently present.  
  **NOTE:** If you already have an `.env` installed, consider replacing it with `.env.sail` as the `DB_*`/`REDIS_*` settings are important for operating inside the sail environment.

```bash
# start the development environment
./sail up -d

# continue with setting up the backend
./sail artisan db:create
./sail artisan migrate --seed
./sail artisan key:generate
./sail artisan admin:create

# and the frontend
./sail yarn
./sail yarn development

# stop the development environment
./sail stop
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,39 @@ When working on new features, keep in mind this website only supports
 
 Internet Explorer isn't supported.
 
+### Development environment
+
+This project uses [Laravel Sail](https://laravel.com/docs/sail) to supply you with a docker
+based development environment. Please take a moment to familiarize yourself with its concepts.
```
Comment (Calinou) [inline README.md:30]:
```suggestion
This project uses [Laravel Sail](https://laravel.com/docs/sail) to supply you with a Docker-based
development environment. Please take a moment to familiarize yourself with its concepts.
```

**Comment by Calinou** [comment]
Could the server be started on port 8080? Port 80 is taken on many development machines (including mine).

**Comment by steffendietz** [comment]
> Could the server be started on port 8080? Port 80 is usually taken by other services on many development machines (including mine).

Absolutely. Sail honors Laravels `APP_PORT` environment setting. I added it to the `.env.sail` and updated the README to point to the http://localhost:8080 entrypoint.

**Comment by steffendietz** [comment]
On a different note.

I just saw that you (@Calinou) [made a comment](https://github.com/Calinou/godot-asset-library-laravel/issues/26#issuecomment-791547705) regarding a **production** docker environment in another issue.

Does this mean by proxy, that the php `7.3` version limitation is about to be discarded? That would be very good news, since `7.3` is EOL, `7.4` is maintenance only and even `8.0` will enter maintenance mode in 4 months.

Seeing as this is not even in production yet, it would be fortunate if you could drop `7.3` support and move to `8.1`.
But thats a different topic I guess. I was just curious.

**Comment by Calinou** [comment]
> I just saw that you (@Calinou) https://github.com/Calinou/godot-asset-library-laravel/issues/26#issuecomment-791547705 regarding a production docker environment in another issue.

We still haven't moved to a new host, so we don't know if we'll be able to run PHP versions newer than 7.3 anytime soon. Therefore, I'd stick to 7.3 for now. LTS Linux distributions really aren't motivated to provide recent PHP versions, unfortunately.

**Comment by steffendietz** [comment]
So. Is there something still missing? Or could this be a reasonable integration of a dev environment?

File: README.md [hunk 2]
```diff
@@ -24,6 +24,43 @@ When working on new features, keep in mind this website only supports
...
+
+It also asks whether it should
+- [install the composer dependencies using a container](https://laravel.com/docs/sail#installing-composer-dependencies-for-existing-projects),
+  if not finding the sail command.
+- install `.env.sail` as the current `.env`, if none is currently present.  
+  **NOTE:** If you already have an `.env` installed, consider replacing it with `.env.sail`
+  as the `DB_*`/`REDIS_*` settings are important for operating inside the sail environment.
+
+```bash
+# start the development environment
```
Comment (Calinou) [inline README.md:43]:
```suggestion
# Start the development environment
```

File: README.md [hunk 3]
```diff
@@ -24,6 +24,43 @@ When working on new features, keep in mind this website only supports
...
+./sail up -d
+
+# continue with setting up the backend
```
Comment (Calinou) [inline README.md:46]:
```suggestion
# Continue with setting up the backend
```

File: README.md [hunk 4]
```diff
@@ -24,6 +24,43 @@ When working on new features, keep in mind this website only supports
...
+./sail artisan db:create
+./sail artisan migrate --seed
+./sail artisan key:generate
+./sail artisan admin:create
+
+# and the frontend
```
Comment (Calinou) [inline README.md:52]:
```suggestion
# ...and the frontend
```

File: README.md [hunk 5]
```diff
@@ -24,6 +24,43 @@ When working on new features, keep in mind this website only supports
...
+./sail yarn
+./sail yarn development
+```
+
+The development environment will be available at http://localhost:8080 by default.
+
+```bash
+# stop the development environment
```
Comment (Calinou) [inline README.md:60]:
```suggestion
# Stop the development environment
```

**Review by Calinou — APPROVED** [review]

**Pull request merged**